src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultModelBindingContext.cs

```diff
@@ -270,8 +269,15 @@ public override IValueProvider ValueProvider
         /// <inheritdoc />
         public override ValidationStateDictionary ValidationState
         {
-            get { return _state.ValidationState; }
-            set { _state.ValidationState = value; }
+            get { return _validationState; }
+            set
```

Comment:
Looks like this setter exists only for tests, when used together with the test-only parameterless constructor. Please add comments because it's too easy to mess the non-stacked property values up if the assumptions aren't stated.